jps-plugin/build.gradle

```diff
@@ -1,6 +1,6 @@
 jar.archiveFileName = "thrift-jps.jar"
 
 java {
-    sourceCompatibility = JavaVersion.VERSION_1_8
-    targetCompatibility = JavaVersion.VERSION_1_8
-}
\ No newline at end of file
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
```

Comment:
Using 1.8 would generate the following warnings, so I think we could update to 17?
```
> Task :jps-plugin:verifyPluginConfiguration
[gradle-intellij-plugin :jps-plugin jps-plugin:jps-plugin:verifyPluginConfiguration] The following plugin configuration issues were found:
- The Java configuration specifies sourceCompatibility=1.8 but IntelliJ Platform 2023.3 requires sourceCompatibility=17.
See: https://jb.gg/intellij-platform-versions
```